### PR TITLE
Add misc specification

### PR DESCRIPTION
**Pull request by pablo-gar**
Adds specification for dataset-level metadata to the Seurat encoding

Solves 
#141

### CODE REVIEW

File: cellxgene_schema_cli/tests/questions_workshop.md [hunk 1]
```diff
@@ -0,0 +1 @@
+1.
```
Comment (brianraymor) [inline cellxgene_schema_cli/tests/questions_workshop.md:1]: What is this?

Comment (pablo-gar) [reply cellxgene_schema_cli/tests/questions_workshop.md:1]: hm I don't know what ended up there, I'll delete it

File: schema/2.0.0/seurat_encoding.md [hunk 2]
```diff
@@ -237,4 +237,64 @@ The matrix with reductions is stored in the `cell.embeddings` slot of `SeuratObj
...
+      <td><code>seurat_object@misc</code></td>
+    </tr>
+    <tr>
+      <th>Value</th>
+        <td><code>list[character|list]</code></td>
+    </tr>
+</tbody></table>
+<br>
+
+Only fields defined in the [schema](./schema.md/#uns-dataset-metadata) are transferred from the AnnData to the Suerat object:
```
Comment (brianraymor) [inline schema/2.0.0/seurat_encoding.md:254]: Suerat -> Seurat

**Comment by brianraymor** [comment]
Missed this earlier, can it be corrected? - Seurat allows elements of **assasy** to have any arbitrary name

File: schema/2.0.0/seurat_encoding.md [hunk 3]
```diff
@@ -237,4 +237,64 @@ The matrix with reductions is stored in the `cell.embeddings` slot of `SeuratObj
 
 ### Dataset Metadata
 
-At the moment a Seurat object downloaded from cellxgene does not contain the dataset metadata stored in the AnnData slot `uns`.
+The dataset metadata from the AnnData slot `uns` is stored in the Seurat object in the slot `misc`.
```
Comment (brianraymor) [inline schema/2.0.0/seurat_encoding.md:240]: This should follow a similar style to your previous text and not include references to AnnData.

Dataset metadata is stored in the `misc` slot. 

and

Only fields defined in the [schema](./schema.md/#uns-dataset-metadata) are transferred to the Seurat object

**Comment by brianraymor** [comment]
@metakuni - since you're implementing this feature, you should review this PR to ensure that it addresses any questions that you may have.

File: schema/2.0.0/seurat_encoding.md (hunk 3 above)
Comment (metakuni) [reply schema/2.0.0/seurat_encoding.md:240]: Do we need/want to include a statement about why `uns` is being stored in `misc`? i.e.:
* https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/single-cell-curation/149#issuecomment-934641155

**Review by metakuni — APPROVED** [review]

**Comment by brianraymor** [comment]
> Do we need/want to include a statement about why uns is being stored in misc? i.e.:

Generally, design rationales/discussions are not captured in this type of specification - only the decisions.

File: schema/2.0.0/seurat_encoding.md [hunk 4]
```diff
@@ -237,4 +237,64 @@ The matrix with reductions is stored in the `cell.embeddings` slot of `SeuratObj
...
+      <td><code>character</code></td>
+      <td>No</td>
+    </tr>
+    <tr>
+      <td>X_normalization</td>
+      <td><code>character</code></td>
+      <td>No</td>
+    </tr>
+    <tr>
+      <td>feature_name</td>
```
Comment (metakuni) [inline schema/2.0.0/seurat_encoding.md:279]: Just notice that `feature_name` isn't listed in:
* https://github.com/chanzuckerberg/single-cell-curation/blob/main/schema/2.0.0/schema.md#uns-dataset-metadata

Should it be?

Comment (pablo-gar) [reply schema/2.0.0/seurat_encoding.md:279]: Thanks for catching this, it's an error

**Pull request merged**